ui/components/src/LineChart/utils.ts

```diff
@@ -75,17 +76,26 @@ export function getFormattedDate(value: number, rangeMs: number, timeZone?: stri
     },
     timeZone
   );
+  let roundToNearestFiveMinutes = true;
   const thirtyMinMs = 1800000;
   const dayMs = 86400000;
   if (rangeMs <= thirtyMinMs) {
     dateFormatOptions.second = 'numeric';
+    roundToNearestFiveMinutes = false;
   } else if (rangeMs >= dayMs) {
     dateFormatOptions.month = 'numeric';
     dateFormatOptions.day = 'numeric';
   }
   const DATE_FORMAT = new Intl.DateTimeFormat(undefined, dateFormatOptions);
+
+  if (roundToNearestFiveMinutes === true) {
+    // https://date-fns.org/v2.29.3/docs/roundToNearestMinutes
+    const roundedValue = roundToNearestMinutes(Number(value), { nearestTo: 5, roundingMethod: 'round' });
+    return DATE_FORMAT.format(roundedValue).replace(/, /g, ' ');
+  }
+
   // remove comma when month / day present
-  return DATE_FORMAT.format(value).replace(/, /g, ' ');
+  return DATE_FORMAT.format(Number(value)).replace(/, /g, ' ');
```

Comment:
Pulled down the PR and played with this branch a little

One thing I realized is that this means that the underlying, actual timestamp will be different from the x-axis label, because of the rounding 

So, when I hover on a vertical x-axis tick, the x-axis label says "4:45", but then the cursor reveals that the actual timestamp is "4:42:45"

I think if I was a user, and the time reflected by the x-axis label and the time reflected by the cursor were not in sync, I would think something was off 

It seems like we would want to try to show the actual data at the actual time, though I'm not sure how feasible that is 🤔 In other words, if the x-axis label is "4:45", then try to show the data that corresponds to 4:45...

Or.... just update the tooltip with the same rounding, but that seems like could really lead to some bad issues for users who are debugging and need actual, truthful timestamps...

Reply:
Anyway, excited for this improvement, thanks for grabbing it

Reply:
All good points, yeah I wasn’t 100% sure about this solution and you articulated it really well why, need to keep thinking on this 🫤